pwiz_tools/Skyline/Util/LazyComboBox.cs

```diff
@@ -0,0 +1,141 @@
+﻿/*
+ * Original author: Brian Pratt <bspratt .at. proteinms dot net>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ *
+ * Copyright 2021 University of Washington - Seattle, WA
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Windows.Forms;
+using pwiz.Skyline.Properties;
+
+namespace pwiz.Skyline.Util
+{
+    /// <summary>
+    /// Substitute for ComboBox which avoids performance issues by only
+    /// creating its dropdown list when user action demands it
+    /// </summary>
+    public class LazyComboBox : Button
```

Comment:
I wonder whether this class should be named "LazyDropDownList" since that's the only combo box style it supports. (Actually. maybe just call it "DropDownList" since "Lazy" is too judgmental)
Also, maybe move this class into "Common\Controls" folder.